src/pages/index.js

```diff
@@ -0,0 +1,321 @@
+const avatarForm = avatarModal.querySelector(".modal__form");
+const avatarSubmitButton = avatarModal.querySelector(".modal__submit-btn");
+const avatarInput = avatarModal.querySelector("#profile-avatar-input");
+const deleteModal = document.querySelector("#delete-modal");
+const deleteForm = deleteModal.querySelector(".modal__form-delete");
+const deleteModalButton = deleteModal.querySelector(
+  ".modal__submit-btn-delete"
+);
+
+const deleteModalCancelButton = deleteModal.querySelector(
+  ".modal__submit-btn-cancel"
+);
+
+function getCardElement(data) {
+  console.log("Creating card with data:", data);
+
+  const cardElement = cardTemplate.content
+    .querySelector(".card")
+    .cloneNode(true);
+
+  const cardNameEl = cardElement.querySelector(".card__title");
+  const cardImageEl = cardElement.querySelector(".card__image");
+  const cardLikeButton = cardElement.querySelector(".card__like-button");
+  const cardDeleteButton = cardElement.querySelector(".card__delete");
+
+  cardNameEl.textContent = data.name;
+  cardImageEl.src = data.link;
+  cardImageEl.alt = data.name;
+  if (data.isLiked) {
+    cardLikeButton.classList.add("card__like-button_liked");
+  }
+  cardImageEl.addEventListener("click", () => {
+    openModal(previewModal);
+    previewModalImageElement.src = data.link;
+    previewModalImageElement.alt = data.name;
+    previwModalCaptionElement.textContent = data.name;
+  });
+
+  cardLikeButton.addEventListener("click", (evt) => handleLike(evt, data._id));
+  console.log("Like button initial state:", {
+    cardId: data._id,
+    isLiked: data.isLiked,
+    hasLikedClass: cardLikeButton.classList.contains("card__like-button_liked"),
+  });
+  cardDeleteButton.addEventListener("click", () => {
+    handleDeleteCard(cardElement, data._id);
+  });
+
+  return cardElement;
+}
+
+function openModal(modal) {
+  modal.classList.add("modal_opened");
+
+  const closeModalOnEscape = (evt) => {
+    if (evt.key === "Escape") {
+      closeModal(modal);
+      document.removeEventListener("keydown", closeModalOnEscape);
+    }
+  };
+  document.addEventListener("keydown", closeModalOnEscape);
+}
+
+function closeModal(modal) {
+  modal.classList.remove("modal_opened");
+}
+
+function handleEditFormSubmit(evt) {
+  evt.preventDefault();
+  const submitBtn = evt.submitter;
+  submitBtn.textContent = "Saving...";
+  console.log("Form submitted");
+  api
+    .editUserInfo({ name: nameInput.value, about: descriptionInput.value })
+    .then((data) => {
+      profileName.textContent = data.name;
+      profileDescription.textContent = data.about;
+      closeModal(editModal);
+      evt.target.reset();
+      disableButton(
+        editFormElement.querySelector(".modal__submit-btn"),
+        settings
+      );
+    })
+    .catch(console.error)
+    .finally(() => {
+      submitBtn.textContent = "Save";
+    });
+}
+
+function handleAddCardFormSubmit(evt) {
+  console.log("Form submitted");
```

Comment:
* According to the task, you need to change submit buttons text (`Saving…`) while waiting for the server response. And the user should see it.